### PR TITLE
Various enhancements

### PR DESCRIPTION
**Pull request by regislebrun**
The main contributions are:
+ the ability to provide a global covariance error in BLUE and ThreeDVAR, to cope with situations where the error has a varying covariance or where there is a correlation between the errors
+ the visual test of fit between a distribution and a sample set
+ the ability to draw the survival function for distributions
+ the ability to specify if the discretization has to be done on a linear grid or a logarithmic grid for the various distribution graphs

### CODE REVIEW

File: lib/src/Uncertainty/Model/Distribution.cxx [hunk 1]
```diff
@@ -28,8 +28,8 @@ BEGIN_NAMESPACE_OPENTURNS
 CLASSNAMEINIT(Distribution)
 
 /* Default constructor */
-Distribution::Distribution()
-  : TypedInterfaceObject<DistributionImplementation>(new Uniform())
+  Distribution::Distribution()
```
Comment (sofianehaddad) [inline lib/src/Uncertainty/Model/Distribution.cxx]: fix indentation please

File: lib/src/Uncertainty/Model/Distribution.cxx [hunk 2]
```diff
@@ -418,7 +418,7 @@ Point Distribution::computeInverseSurvivalFunction(const Scalar prob) const
 }
 
 Point Distribution::computeInverseSurvivalFunction(const Scalar prob,
-    Scalar & marginalProb) const
+                                                   Scalar & marginalProb) const
 {
```
Comment (sofianehaddad) [inline lib/src/Uncertainty/Model/Distribution.cxx]: same here

File: lib/src/Uncertainty/Model/Distribution.cxx [hunk 3]
```diff
@@ -561,8 +561,8 @@ Sample Distribution::computeCDF(const Scalar xMin,
 }
 
 Sample Distribution::computeComplementaryCDF(const Scalar xMin,
-    const Scalar xMax,
-    const UnsignedInteger pointNumber) const
+                                             const Scalar xMax,
+                                             const UnsignedInteger pointNumber) const
```
Comment (sofianehaddad) [inline lib/src/Uncertainty/Model/Distribution.cxx]: same here

File: lib/src/Uncertainty/Model/Distribution.cxx [hunk 4]
```diff
@@ -632,7 +632,7 @@ Sample Distribution::computeQuantile(const Point & prob,
 }
 
 Scalar Distribution::computeScalarQuantile(const Scalar prob,
-    const Bool tail) const
+                                           const Bool tail) const
```
Comment (sofianehaddad) [inline lib/src/Uncertainty/Model/Distribution.cxx]: here too

File: lib/src/Uncertainty/Model/Distribution.cxx [hunk 5]
```diff
@@ -672,14 +672,14 @@ Interval Distribution::computeBilateralConfidenceIntervalWithMarginalProbability
...
+   for all i\in{1,...,d}, a_i=-\intfy and P(X_i<=b_i)=\beta
    if upper == true
-     for all i\in{1,...,d}, P(a_i<=X_i)=\beta and b_i=\intfy
+   for all i\in{1,...,d}, P(a_i<=X_i)=\beta and b_i=\intfy
 
    where in both cases \beta is such that P(X\in\prod_{i=1}^d[a_i, b_i])=p
 */
 Interval Distribution::computeUnilateralConfidenceInterval(const Scalar prob,
-    const Bool tail) const
+                                                           const Bool tail) const
```
Comment (sofianehaddad) [inline lib/src/Uncertainty/Model/Distribution.cxx]: here too

File: lib/src/Uncertainty/Model/Distribution.cxx [hunk 6]
```diff
@@ -806,7 +806,7 @@ Scalar Distribution::computeDensityGeneratorSecondDerivative(const Scalar betaSq
 
 /* Compute the radial distribution CDF */
 Scalar Distribution::computeRadialDistributionCDF (const Scalar radius,
-    const Bool tail) const
+                                                   const Bool tail) const
```
Comment (sofianehaddad) [inline lib/src/Uncertainty/Model/Distribution.cxx]: here too

**Comment by regislebrun** [comment]
The failing test is due to a timeout in object_pickle. Not clear to me if I can do anything here...

**Comment by jschueller** [comment]
good for you @sofianehaddad ?

**Pull request merged**